### ISSUE TITLE
E1101 (no-member) is not suppressed when we know it should be

### ISSUE THREAD

**Issue by endafarrell** [nelfin/pylint-protobuf#2]
When running ``pylint --load-plugins pylint_protobuf --errors-only Transform/RIB_2/src/rib_search_2_transform/tx_places.py`` on a file that uses protobuf-generated classes I get the following unexpected error:

``
************* Module LG.Transform.RIB_2.src.rib_search_2_transform.tx_places
Transform/RIB_2/src/rib_search_2_transform/tx_places.py:111:30: E1101: Instance of 'PlacesPartition' has no 'partition_name' member (no-member)
``
The class 'PlacesPartition' does have a field called `partition_name`:

```:python
_PLACESPARTITION = _descriptor.Descriptor(
  name='PlacesPartition',
  full_name='com.enda.schema.rib.v2.PlacesPartition',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    _descriptor.FieldDescriptor(
      name='partition_name', full_name='com.enda.schema.rib.v2.PlacesPartition.partition_name', index=0,
      number=1, type=9, cpp_type=9, label=1,
      has_default_value=False, default_value=_b("").decode('utf-8'),
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      serialized_options=None, file=DESCRIPTOR),
 ...
```
(there's a lot more, but this is the interesting bit).

Any ideas what's going on? It seems as though none of the protobuf fields are being recognised.

I'm running in py3.7.2 and there's one thing that may be affecting this plugin - the generated pb2.py files have all had their package names prefixed with ``Transform.RIB_2.src.`` so that they will fit into our build structures (we cannot keep the dir hierarchy flat as they do in google. I _suspect_ that this is affecting the module names - is that likely? Do you happen to have any recommendations on how I might try to work through this problem we have?

**Comment by nelfin** [comment]
Hi @endafarrell, thanks for trying out the code and the well written issue. From the looks of it, it's not any issue with package renaming or the like, it seems more like that this plugin isn't *suppressing* errors that it knows are erroneous. `E1101` is actually a built-in lint: `pylint-protobuf` only raises [`E5901 (protobuf-undefined-attribute)`](https://github.com/nelfin/pylint-protobuf/blob/57a6fdbf23d62322cfb10575584bb93af0bf51a0/pylint_protobuf/__init__.py#L5).

I'll work on fixing that, thanks for pointing it out.

**Comment by zapstar** [comment]
Unless we can prevent the `E1101` from being thrown by PyLint this plugin is not really usable in production. We cannot just disable `no-member`as it would lead to more bugs in code unrelated to protobuf.

**Comment by zapstar** [comment]
Also please consider editing the title to something that reflects `no-member` is still raised by PyLint.

**Comment by nelfin** [comment]
@zapstar, good point. I've been working on a test to check the integration of these and I have some ideas about how to suppress the warnings programatically that I'll try once that's working in the next few days.

**Comment by nelfin** [comment]
I've released a fix for this as `0.3.dev3`, this should get included in the `0.3` when that goes up.

**Comment by endafarrell** [comment]
@endafarrell Do you have an ETA on when 0.3 will be released?

**Comment by nelfin** [comment]
@endafarrell, no time like the present. I've uploaded 0.3 now and removed the assert that was causing that message you saw.

**Comment by endafarrell** [comment]
@nelfin I've just now tried the new 0.3: first impressions look good. I'll give a more complete update over the next few days.

**Comment by endafarrell** [comment]
@nelfin I'm liking the new 0.3 - it works quite well for me. I can run it on modules which were protpbuf-generated, though if I leave the plugin on for a pylint over all of the codebase it does have some stderr output:

```
internal error with sending report for module ['API/src/blueprints/demo_201810.py']
list index out of range
internal error with sending report for module ['API/src/utils/convert.py']
'method' object is not iterable
internal error with sending report for module ['API/src/services/graph.py']
'method' object is not iterable
internal error with sending report for module ['Common/s3.py']
'method' object is not iterable
internal error with sending report for module ['Common/environment.py']
'method' object is not iterable
internal error with sending report for module ['Common/utils/utils.py']
'method' object is not iterable
internal error with sending report for module ['Common/utils/category_mapping.py']
No inference function for <Compare l.213 at 0x118b81780>.
```

There are also these two stderr outputs from modules which directly use _pb2 modules:
```

internal error with sending report for module ['Transform/X/src/transform/street_names.py']
Inference failed for <Subscript l.70 at 0x11d824390>.
internal error with sending report for module ['Transform/X/src/tx_places.py']
Inference failed for <Call l.301 at 0x120890438>.
```

**Comment by nelfin** [comment]
Hey @endafarrell,

Thanks for testing this out. This might be related to issue #5 raised just
now. I'll check both out today.
